data-serving/reusable-data-service/reusable_data_service/model/case.py

```diff
@@ -0,0 +1,46 @@
+import dataclasses
+import datetime
+import json
+
+
+@dataclasses.dataclass()
+class DayZeroCase:
+    """This class implements the "day-zero" data schema for Global.health.
+    At the beginning of an outbreak, we want to collect at least this much
+    information about an individual case for the line list.
+
+    Parameters here are defined to be keyword-only and not set in the
+    initialiser, so that clients can use Builder to populate them. Use
+    the validate() method to determine whether an instance is in a
+    consistent state (this also means we can add custom validation logic
+    to that function)."""
+
+    _: dataclasses.KW_ONLY
+    confirmation_date: datetime.datetime = dataclasses.field(init=False)
+
+    @classmethod
+    def from_json(cls, obj: str) -> type:
+        """Create an instance of this class from a JSON representation."""
+        case = cls()
+        source = json.loads(obj)
+        for key in source:
+            if key in ["confirmation_date"]:
+                # parse as an ISO 8601 date
+                date = datetime.datetime.strptime(source[key], "%Y-%m-%dT%H:%M:%S.%fZ")
```

Comment:
confirmation_date should not have time, unless we want to support that

Reply:
Yup good point. I parse this as a date time still to handle the timezone correctly, then get a date from it.